src/transformers/models/megatron_gpt2/convert_megatron_gpt2_checkpoint.py

```diff
@@ -0,0 +1,238 @@
+
+        # Copy the bias.
+        elif weight_or_bias == "bias":
+
+            out_name = megatron_to_transformers[op_name]
+            output_state_dict[layer_name + out_name + "bias"] = val
+
+    # The final layernorm.
+    output_state_dict["transformer.ln_f.weight"] = transformer["final_layernorm.weight"]
+    output_state_dict["transformer.ln_f.bias"] = transformer["final_layernorm.bias"]
+
+    # For LM head, transformers' wants the matrix to weight embeddings.
+    output_state_dict["lm_head.weight"] = word_embeddings
+
+    # The config.
+    output_config = {
+        "activation_function": "gelu_new",
+        "architectures": ["GPT2LMHeadModel"],
+        "attn_pdrop": 0.1,
+        "bos_token_id": 50256,
+        "embd_pdrop": 0.1,
+        "eos_token_id": 50256,
+        "initializer_range": 0.02,
+        "layer_norm_epsilon": 1e-05,
+        "model_type": "gpt2",
+        "n_ctx": 1024,
+        "n_embd": 1024,
+        "n_head": 16,
+        "n_layer": 24,
+        "n_positions": 1024,
+        "resid_pdrop": 0.1,
+        "summary_activation": None,
+        "summary_first_dropout": 0.1,
+        "summary_proj_to_labels": True,
+        "summary_type": "cls_index",
+        "summary_use_proj": True,
+        "vocab_size": 50257,
+    }
+
+    # It should be done!
+    return output_state_dict, output_config
+
+
+####################################################################################################
+
+
+def main():
+    # Create the argument parser.
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--print-checkpoint-structure", action="store_true")
+    parser.add_argument("path_to_checkpoint", type=str, help="Path to the ZIP file containing the checkpoint")
+    args = parser.parse_args()
+
+    # Extract the basename.
+    basename = os.path.splitext(args.path_to_checkpoint)[0]
+
+    # Load the model.
+    print('Extracting PyTorch state dictionary from "{}"'.format(args.path_to_checkpoint))
+    with zipfile.ZipFile(args.path_to_checkpoint, "r") as checkpoint:
+        with checkpoint.open("release/mp_rank_00/model_optim_rng.pt") as pytorch_dict:
+            input_state_dict = torch.load(pytorch_dict, map_location="cpu")
+
+    # Convert.
+    print("Converting")
+    output_state_dict, output_config = convert_megatron_checkpoint(args, input_state_dict)
+
+    # Print the structure of converted state dict.
+    if args.print_checkpoint_structure:
+        recursive_print(None, output_state_dict)
+
+    # Store the config to file.
+    output_config_file = basename + "_config.json"
+    print('Saving config to "{}"'.format(output_config_file))
+    with open(output_config_file, "w") as f:
+        json.dump(output_config, f)
+
+    # Store the state_dict to file.
+    output_checkpoint_file = basename + "_checkpoint.pt"
+    print('Saving checkpoint to "{}"'.format(output_checkpoint_file))
+    torch.save(output_state_dict, output_checkpoint_file)
```

Comment:
Same remark regarding the filenames!

Reply:
Same answer. I'll do it as soon as you tell me what you think I should use for `<PATH>` :)